src/pods/toolbar/components/export-button/export-button.business.ts

```diff
@@ -43,3 +43,46 @@ export const expandAllFieldsInTables = (table: TableVm[]) =>
     ...table,
     fields: expandTableFields(table.fields),
   }));
+
+// Export Schema functions
+const getPropertyJsonSchema = (field: FieldVm): string => {
+  if (field.isArray) {
+    return `"${field.name}": { bsonType: "array", items: { bsonType: "${field.type}" } }`;
+  }
+
+  if (field.children && field.children.length > 0) {
+    const properties = getPropertiesJsonSchema(field.children, false);
+    return `"${field.name}": { bsonType: "object", title: "${field.name}", properties: { ${properties}, }, }`;
+  }
+  return `"${field.name}": { bsonType: "${field.type}" }`;
+};
+
+const getPropertiesJsonSchema = (fields: FieldVm[], useTab = true): string => {
+  const separator = useTab ? ',\n\t\t\t\t' : ', ';
+  return fields.map(getPropertyJsonSchema).join(separator);
+};
+
+// TODO -> Implement this function, when NN feature is implemented
+// const getRequiredFields = (fields: FieldVm[]): string[] => {}
+
+const getSchemaScriptFromTableVm = (table: TableVm): string => {
+  const properties = getPropertiesJsonSchema(table.fields);
+  const schemaScript = `db.createCollection("${table.tableName}", {
+  validator: {
+    $jsonSchema: {
+      bsonType: "object",
+      title: "${table.tableName}",
+      required: [],
+      properties: {
+        ${properties},
+      },
+    },
+  },
+});`;
+
+  return schemaScript;
+};
+
+export const getSchemaScriptFromTableVmArray = (tables: TableVm[]): string => {
```

Comment:
I'm going to add an issue in github to implement unit tests on this function and reference , it can be a good entry point for students